### DISCUSSION TITLE
Does Xournal++ preserve all of the original PDF's formatting upon export to a new PDF?

### DISCUSSION THREAD

**Discussion by jtherrmann** [xournalpp/xournalpp#5429] [Q&A] [answered]
If you open a PDF with Xournal++, make some modifications (e.g. adding annotations, deleting pages, etc.), and then export to a new PDF, is there any possibility that Xournal++ will modify the original PDF's formatting (e.g. font family, font size, text layout) when exporting to the new PDF?

For example, LibreOffice Draw breaks PDF formatting when editing and then exporting a PDF document, as described [here](https://ask.libreoffice.org/t/pdf-editing-capabilities-in-free-writer/83997/2).

I would like to be confident that editing and then exporting a PDF using Xournal++ will preserve all of the original formatting, so that the new PDF renders exactly the same as the original (except for whatever annotations were added using Xournal++, of course).

**Comment by rolandlo** [comment, marked as answer]
Visually the formatting in the background PDF and the exported PDF looks the same in my experience, but Xournal++ uses Poppler/Cairo to render the background PDF to a cairo surface and things like PDF forms are not preserved.
You can test without adding any modifications. The exported PDF will not be the same as the background PDF. The same flattening is also applied e.g. by pdftocairo, see e.g. [this stackexchange article](https://tex.stackexchange.com/questions/443818/how-do-i-flatten-or-render-a-pdf-form-so-it-can-be-included-in-another-pdf-with) for some more infos.

**Reply by jtherrmann** [reply]
Thanks for your response. For what it's worth, I tested Xournal++ 1.1.1 on Pop!_OS 22.04 LTS with a random PDF, and the exported PDF appears very close to the original, except for some minor differences in spacing between text characters. On the other hand `pdftk` (installed via `apt`) seems to preserve the original PDF formatting perfectly, including spacing between characters. So, I will probably use `pdftk` for extracting pages when preserving formatting is important, and continue using Xournal++ for annotations.
